src/Makefile

```diff
@@ -76,9 +76,21 @@ PUBLIC_HEADERS += $(LIBUV_INC)/uv*
 endif
 PUBLIC_HEADER_TARGETS := $(addprefix $(build_includedir)/julia/,$(notdir $(PUBLIC_HEADERS)))
 
-# In LLVM < 3.4, --ldflags includes both options and libraries, so use it both before and after --libs
-# In LLVM >= 3.4, --ldflags has only options, and --system-libs has the libraries.
+ifeq ($(LLVM_VER_SHORT),$(filter $(LLVM_VER_SHORT),3.3 3.4 3.5 3.6 3.7))
+# `llvm-config` support `--link-*` only on 3.8+
+USE_LLVM_CONFIG_LIB := 0
+else
+ifeq ($(OS),WINNT)
+# Somehow `llvm-config --libs` does not work with shlib build on windows
```

Comment:
I assumed that his is what you were trying to figure out after https://github.com/JuliaLang/julia/pull/19443#issuecomment-268162499 so I didn't make this conditional on windows back then ...... I can wait a few days longer.....